coderd/unhanger/detector_test.go

```diff
@@ -37,7 +43,7 @@ func TestDetectorNoJobs(t *testing.T) {
 		statsCh    = make(chan unhanger.Stats)
 	)
 
-	detector := unhanger.New(ctx, db, pubsub, log, tickCh).WithStatsChannel(statsCh)
+	detector := unhanger.New(ctx, wrapDBAuthz(db, log), pubsub, log, tickCh).WithStatsChannel(statsCh)
```

Comment:
Feels odd to have a raw `db` laying around outside this function call scope. I'd probably do this to not have a `db` laying around. Up to you :man_shrugging: 

```golang
	var (
		ctx         = testutil.Context(t, testutil.WaitLong)
		rdb, pubsub = dbtestutil.NewDB(t)
		db          = wrapDBAuthz(rdb, slogtest.Make(t, nil))
		log         = slogtest.Make(t, nil)
		tickCh      = make(chan time.Time)
		statsCh     = make(chan unhanger.Stats)
	)
```

Reply:
The annoying thing about that is that I'd have to set up a dbauthz principal for the calls that write the job, user, template, etc.  Authz for those operations is irrelevant to this test.

Reply:
Ah, yea it is annoying. `dbgen` gets around that with a `genCtx`: https://github.com/coder/coder/blob/40bf367b5de6f19c9fbe0e0cf854d70f17bdceb0/coderd/database/dbgen/dbgen.go#L35-L35

If you use dbgen to handle all the initial state, you can ignore that requirement.

Reply:
Almost, but not quite all state is `dbgen`.  It make sense to me to keep this way, since it really is just the unhanger under test here, so only it need the dbauthz.